Tempura/Core/ViewController.swift

```diff
@@ -226,15 +226,25 @@ open class ViewController<V: ViewControllerModellableView & UIView>: UIViewContr
   /// Override to setup something after init.
   open func setup() {}
   
-  /// Shortcut to the dispatch function.
-  @available(*, deprecated, message: "remove `action` label")
-  open func dispatch(action: Action) {
-    self.store.dispatch(action)
+  /// Shortcut to the non-generic dispatch function.
+  open func dispatch(_ dispatchable: Dispatchable) {
+    self.store.dispatch(dispatchable)
   }
   
   /// Shortcut to the dispatch function.
+  open func dispatch<T: Dispatchable>(_ dispatchable: T) {
+    _ = self.store.dispatch(dispatchable)
+  }
```

Comment:
This is not needed as, given there are no return values, it's not possible to invoke this method if `dispatch(_ dispatchable: Dispatchable)` is already defined.

This is a small playground to prove it.
```swift
protocol State {
  init()
}


class Promise<Value> {
  
}

public protocol Dispatchable {}

protocol AnySideEffect: Dispatchable {}

protocol SideEffect: AnySideEffect {
  associatedtype ReturnValue
}

protocol AnyStateUpdater: Dispatchable {}

protocol StateUpdater: AnyStateUpdater {
  associatedtype StateType: State
}


struct SE1: SideEffect {
  typealias ReturnValue = Int
}

struct SU1: StateUpdater {
  typealias StateType = AppState
}

protocol AnyStore: class {
  var anyState: State { get }
  
  @discardableResult
  func dispatch<T: Dispatchable>(_ dispatchable: T) -> Promise<Void>
  
  @discardableResult
  func dispatch<T: SideEffect>(_ dispatchable: T) -> Promise<T.ReturnValue>
  
  @discardableResult
  func dispatch(_ dispatchable: Dispatchable) -> Promise<Any>
}


class PartialStore<S: State>: AnyStore {
  var state: S
  
  var anyState: State {
    return self.state
  }
  
  init(state: S) {
    self.state = state
  }
  
  @discardableResult
  func dispatch<T: Dispatchable>(_ dispatchable: T) -> Promise<Void> {
    print("#1")
    fatalError()
  }
  
  @discardableResult
  func dispatch<T: SideEffect>(_ dispatchable: T) -> Promise<T.ReturnValue> {
    print("#2")
    fatalError()
  }
  
  @discardableResult
  func dispatch(_ dispatchable: Dispatchable) -> Promise<Any> {
    print("#3")
    fatalError()
  }
}

struct AppState: State {}

let store = PartialStore<AppState>(state: AppState())

//store.dispatch(SE1()) //#3
//store.dispatch(SU1()) //#3

//let p = store.dispatch(SU1()) //#3

//let p: Promise<Void> = store.dispatch(SU1()) //#1

//store.dispatch(SE1()) //#3

//let p: Promise<Int> = store.dispatch(SE1()) //#2

//let p: Promise<Any> = store.dispatch(SE1()) //#3

//let c  = store.dispatch(SU1()) //#3

class VC<S: State> {
  var store: PartialStore<S>
  
  init(store: PartialStore<S>) {
    self.store = store
  }
  
  func dispatch(_ dispatchable: Dispatchable) {
    self.store.dispatch(dispatchable)
  }
  
  func dispatch<T: Dispatchable>(_ dispatchable: T) {
    self.store.dispatch(dispatchable)
  }
}

//let c: Promise<Void> = store.dispatch(SU1()) //#1

let vc = VC<AppState>(store: store)
//vc.dispatch(SU1()) // #3
//vc.dispatch(SE1()) // #3
// here there is no possibility to invoke #1 or #2
```